index.js

```diff
@@ -74,12 +74,24 @@ function build (schema, options) {
     ;
     return ${main}
   `
-  if (schema.additionalProperties === true) {
+  if (hasAdditionalPropertiesTrue(schema)) {
     return (new Function('fastSafeStringify', code))(fastSafeStringify)
   }
   return (new Function(code))()
 }
 
+function hasAdditionalPropertiesTrue (schema) {
```

Comment:
Can also do this:
```js
function hasAdditionalPropertiesTrue (schema) {
  const stringifiedSchema = fastSafeStringify(schema)

  return stringifiedSchema.includes('"additionalProperties":true')
}
```

That safeguards against circular objects, and thus infinite recursion